src/operators/catch.ts

```diff
@@ -50,7 +50,17 @@ class CatchSubscriber<T> extends Subscriber<T> {
     if (result === errorObject) {
       this.destination.error(errorObject.e);
     } else {
-      this.add(result.subscribe(this.destination));
+      this.lastSubscription.unsubscribe();
+      this.lastSubscription = result.subscribe(this.destination);
     }
   }
+
+  _complete() {
+    this.lastSubscription.unsubscribe();
+    this.destination.complete();
+  }
+
+  _unsubscribe() {
```

Comment:
Ah, didn't aware it's inherited from `Subscription`. thought it's inherited from `Subscriber` and tried to look `Subscriber::_unsubscribe()` instead.. my bad. :(